kontrol/kontrol.go

```diff
@@ -128,6 +131,24 @@ func (k *Kontrol) AddAuthenticator(keyType string, fn func(*kite.Request) error)
 	k.Kite.Authenticators[keyType] = fn
 }
 
+// DeleteKeyPair deletes the key with the given id or public key. (One of eac
+// other can be empty)
+func (k *Kontrol) DeleteKeyPair(id, public string) error {
+	if k.keyPair == nil {
+		return errors.New("Key pair storage is not initialized")
+	}
+
+	k.keyPair.DeleteKey(&KeyPair{
+		ID:     id,
+		Public: public,
+	})
+
+	// delete latest added key so it doesn't get picked up
```

Comment:
Because we use a slice and the keys are appended each time we use `AddKeyPair`:

```go
	// set last set key pair
	k.lastPublic = append(k.lastPublic, public)
	k.lastPrivate = append(k.lastPrivate, private)
```